pkg/testmachinery/testdefinition/prepare.go

```diff
@@ -111,60 +118,51 @@ func (p *PrepareDefinition) AddRepositoriesAsArtifacts() error {
 
 // addKubeconfig adds all defined kubeconfigs as files to the prepare pod
 func (p *PrepareDefinition) addKubeconfigs(kubeconfigs tmv1beta1.TestrunKubeconfigs) error {
-
-	if kubeconfigs.Gardener != "" {
-		kubeconfig, err := base64.StdEncoding.DecodeString(kubeconfigs.Gardener)
-		if err != nil {
-			log.Error("Cannot parse gardener config")
-			return err
-		}
-		p.TestDefinition.AddInputArtifacts(argov1.Artifact{
-			Name: "gardener",
-			Path: fmt.Sprintf("%s/gardener.config", testmachinery.TM_KUBECONFIG_PATH),
-			ArtifactLocation: argov1.ArtifactLocation{
-				Raw: &argov1.RawArtifact{
-					Data: string(kubeconfig),
-				},
-			},
-		})
+	if kubeconfigs.Gardener != nil {
+		return p.addKubeconfig("shoot", kubeconfigs.Shoot)
```

Comment:
```suggestion
		if err := p.addKubeconfig("shoot", kubeconfigs.Shoot); err != nil {
		    return err
		}
```
Shouldn't you only return once the prepare definition at the end of the function? Otherwise the function name is misleading, because of the name smiliarity to addKubeconfig function?

Reply:
Done